var/spack/repos/builtin/packages/libmad/package.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Libmad(AutotoolsPackage):
+    """MAD is a high-quality MPEG audio decoder."""
+
+    homepage = "https://www.underbit.com/products/mad/"
+    url      = "https://download.sourceforge.net/project/mad/libmad/0.15.1b/libmad-0.15.1b.tar.gz"
```

Comment:
It looks like you can also download from ftp://ftp.mars.org/pub/mpeg/ and add ftp://ftp.mars.org/pub/mpeg/archive/ as a `list_url`. That might be better, as SourceForge is notoriously bad for things like `spack versions` and `spack checksum`.

Reply:
Hi,

I added the list_url as an alternative.

Reply:
I would replace the `url` too.

Reply:
I replaced the url. I also fixed the list_url, as the most recent version of libmad was not in the archive.